src/pages/slots/SlotConfiguration.tsx

```diff
@@ -1,69 +1,205 @@
+    EditCapacity,
+    SlotsContainer,
+    Title,
+    Subtitle,
+    SpecificSlotContainer,
+    SlotInfoContainer,
+    SlotInfo,
+    TitlesContainer,
+    ActionsContainer,
+    ScreenContainer,
+    MainTitle,
+    SkeletonsContainer,
+    InfoContainer,
+    AnimatedContainer
+} from "./SlotConfiguration.styles";
 import EditIcon from "../../assets/edit-icon.png";
 import EditSlotForm from "./forms/EditCapacityForm";
 import { useRef, useState } from "react";
 import AddIcon from "../../assets/add-icon.svg";
 import Modal from "../../components/modal/Modal";
 import { DaysOfWeek } from "../../utils/DaysOfWeek";
+import ClockIcon from "../../assets/clock-icon.png";
+import DeleteIcon from "../../assets/delete-icon.png";
+import CalendarIcon from "../../assets/calendar-icon.png";
+import CreateSlotForm from "./forms/CreateSlotForm";
+import { useCreateSlotMutation } from "../../app/services/SlotService";
+import useAuthentication from "../../hooks/useAuthentication";
+import toast from "react-hot-toast";
+import { ModalType, type ModalConfig } from "../../utils/SlotsModalsUtils";
 
 function SlotConfiguration() {
 
     const [showModal, setShowModal] = useState<boolean>(false);
 
+    const [modalType, setModalType] = useState<ModalType>();
+
+    const [createSlot] = useCreateSlotMutation();
+
+    const { userId } = useAuthentication()
+
+    const [selectEnglishValue, setSelectEnglishValue] = useState<string>("");
+
+    const [selectSpanishValue, setSelectSpanishValue] = useState<string>("");
+
+    const handleSelectValue = (event: React.ChangeEvent<HTMLSelectElement>) => {
+        const newEnglishValue = event.target.value;
+        const newSpanishValue = event.target.selectedOptions[0].text;
+        setSelectEnglishValue(newEnglishValue);
+        setSelectSpanishValue(newSpanishValue);
+    }
+
     const editCapacityRef = useRef<any>(null);
 
-    const handleConfirmModal = async () => {
+    const createSlotRef = useRef<any>(null);
+
+    const handleEditCapacityModal = async () => {
         const response = await editCapacityRef.current.submitForm();
         if (response) {
             setShowModal(false);
         }
     }
 
-    return (
-        <MainContainer>
-            {showModal && (
-                <Modal
-                    onClose={() => setShowModal(false)}
-                    showButtons={true}
-                    contentRef={editCapacityRef}
-                    onConfirm={handleConfirmModal}
-                >
-                    <EditSlotForm ref={editCapacityRef} />
-                </Modal>
-            )}
-            <TitleContainer>
-                <Title> MIS TURNOS</Title>
-            </TitleContainer>
+
+    const handleCreateSlotModal = async () => {
+        try {
+            const response = await createSlotRef.current.submitForm();
+            if (response) {
+                await createSlot({ dayOfWeek: selectEnglishValue, startTime: response.startTime, userId }).unwrap();
```

Comment:
Para evitar el problema de que el userId puede ser undefined y en la request el userId no puede serlo, podes hacer esto


```suggestion
                await createSlot({ dayOfWeek: selectEnglishValue, startTime: response.startTime, userId: userId! }).unwrap();
```